contrib/buildinfo/src/mill/contrib/BuildInfo.scala

```diff
@@ -13,17 +13,23 @@ trait BuildInfo extends ScalaModule {
 
   def buildInfoObjectName: String = "BuildInfo"
 
-  def buildInfoMembers: T[Map[String, String]] = T {
-    Map.empty[String, String]
-  }
+  def buildInfoMembers: Map[String, Any] = Map.empty[String, Any]
```

Comment:
This seems really type-unsafe.  I really wouldn't want this in unless you wrapped the types in an ADT.

Reply:
I had an at ADT first but found the usability lacking: to support implicit conversion you had to put the type arguments to your map or you had to manually wrap everything yourself.
While the current approach only supports the basic types and for everything else basically relies on `toString´.
Is there a particular scenario you'd be afraid of that could happen where this leads to an issue?